src/SqlPersistence/Config/SqlPersistenceConfig_Connection.cs

```diff
@@ -19,13 +19,23 @@ public static void ConnectionBuilder(this PersistenceExtensions<SqlPersistence>
                 .Set("SqlPersistence.ConnectionBuilder", connectionBuilder);
         }
 
-        internal static Func<DbConnection> GetConnectionBuilder(this ReadOnlySettings settings)
+        internal static Func<DbConnection> GetConnectionBuilder(this ReadOnlySettings settings, Type storageType)
         {
-            if (settings.TryGet("SqlPersistence.ConnectionBuilder", out Func<DbConnection> value))
+            if (settings.TryGet($"SqlPersistence.ConnectionBuilder.{storageType.Name}", out Func<DbConnection> value))
             {
                 return value;
             }
-            throw new Exception("ConnectionBuilder must be defined.");
+            if (settings.TryGet("SqlPersistence.ConnectionBuilder", out value))
+            {
+                return value;
+            }
+            throw new Exception("The connection to the database must be specified using the ConnectionBuilder method.");
```

Comment:
Suggestion:
` throw new Exception($"Couldn't find connection string for {storageType}. The connection to the database must be specified using the ConnectionBuilder method.");`